### PR TITLE
DOC-6930: Code snippets for ObjC

### PR DESCRIPTION
**Pull request by jayahariv**
1. Initialize Listener
2. Start Listener
3. Stop Listener

### CODE REVIEW

File: modules/objc/examples/code-snippets/SampleCodeTest.m [hunk 1]
```diff
@@ -1019,6 +1019,60 @@ - (void) dontTestSaveWithConflictHandler {
...
+    config.tlsIdentity = nil; // Use with anonymous self signed cert
+    config.authenticator = [[CBLListenerPasswordAuthenticator alloc] initWithBlock: ^BOOL(NSString * username, NSString * password) {
+        if ([self isValidCredentials: username password:password]) {
+            return  YES;
+        }
+        return NO;
+    }];
+    
+    listener = [[CBLURLEndpointListener alloc] initWithConfig: config];
+    // tag::init-urllistener[]
```
Comment (jayahariv) [inline modules/objc/examples/code-snippets/SampleCodeTest.m]:
```suggestion
    // end::init-urllistener[]
```

File: modules/objc/examples/code-snippets/SampleCodeTest.m [hunk 2]
```diff
@@ -1019,6 +1019,60 @@ - (void) dontTestSaveWithConflictHandler {
...
+- (void) dontTestListenerStart {
+    NSError* error = nil;
+    CBLURLEndpointListener* listener = nil;
+    
+    // tag::start-urllistener[]
+    BOOL success = [listener startWithError: &error];
+    if (!success) {
+        NSLog(@"Cannot start the listener: %@", error);
+    }
+    // tag::start-urllistener[]
```
Comment (jayahariv) [inline modules/objc/examples/code-snippets/SampleCodeTest.m]:
```suggestion
    // end::start-urllistener[]
```

File: modules/objc/examples/code-snippets/SampleCodeTest.m [hunk 3]
```diff
@@ -1019,6 +1019,60 @@ - (void) dontTestSaveWithConflictHandler {
...
+}
+
+- (void) dontTestListenerStop {
+    CBLURLEndpointListener* listener = nil;
+    
+    // tag::stop-urllistener[]
+    [listener stop];
+    // tag::stop-urllistener[]
```
Comment (jayahariv) [inline modules/objc/examples/code-snippets/SampleCodeTest.m]:
```suggestion
    // end::stop-urllistener[]
```

File: modules/objc/examples/code-snippets/SampleCodeTest.m [hunk 4]
```diff
@@ -1019,6 +1019,60 @@ - (void) dontTestSaveWithConflictHandler {
...
+    CBLTLSIdentity* identity = nil;
+    // tag::create-self-signed-cert[]
+    
+    NSDictionary* attrs = @{ kCBLCertAttrCommonName: @"Couchbase Inc" };
+    identity = [CBLTLSIdentity createIdentityForServer: YES /* isServer */
+                                            attributes: attrs
+                                            expiration: [NSDate dateWithTimeIntervalSinceNow: 86400]
+                                                 label: @"Server-Cert-Label"
+                                                 error: &error];
+    // tag::create-self-signed-cert[]
```
Comment (jayahariv) [inline modules/objc/examples/code-snippets/SampleCodeTest.m]:
```suggestion
    // end::create-self-signed-cert[]
```

**Review by ibsoln — APPROVED** [review]
Thanks @jayahariv

**Pull request merged**